lib/manageiq/graphql/types/provider.rb

```diff
@@ -0,0 +1,31 @@
+module ManageIQ
+  module GraphQL
+    module Types
+      Provider = ::GraphQL::ObjectType.define do
+        name "Provider"
+        description "A provider is a server with software to manage multiple virtual machines that reside on multiple hosts"
+        interfaces [Taggable]
+
+        global_id_field :id
+        field :database_id,
+              !types.ID,
+              "The database ID of the provider",
+              :property           => :id,
+              :deprecation_reason => "This field may not be included in the ManageIQ Hammer release. Use the global Relay ID ('id') instead."
+        field :api_version, types.String
+        field :created_on, !DateTime
+        field :enabled, types.Boolean
+        field :guid, types.ID
+        field :last_refresh_date, DateTime
+        field :name, !types.String, "The name of the provider"
+        field :type, types.String
```

Comment:
Should we expose the underlying type (which is the model name), or should we be relying on the GraphQL type system directly.  @chrisarcand Thoughts?

Reply:
Whoops, didn't catch that, yes. Until I'm convinced otherwise, *any* field that exposes the usual 'type' column in Rails is wrong.

TBF the whole interface here will have to be rethought to transfer subclasses in active record to actual graphql types that implement higher interfaces, so don't drop everything to remove this now.